app/src/main/java/de/tum/in/tumcampusapp/api/app/TUMCabeClient.java

```diff
@@ -184,12 +184,16 @@ public void addUserToChat(ChatRoom chatRoom, ChatMember member, TUMCabeVerificat
     }
 
     public Observable<ChatMessage> sendMessage(int roomId, TUMCabeVerification verification) {
-        //If the id is zero then its an new entry otherwise try to update it
-        int id = ((ChatMessage) verification.getData()).getId();
-        if (id == 0) {
+        ChatMessage message = (ChatMessage) verification.getData();
+        if (message == null) {
```

Comment:
Ever hear of: https://docs.oracle.com/javase/7/docs/api/java/lang/ClassCastException.html

Reply:
Couldn’t it still be null if no `data` has been set?

Reply:
Practically yes, but then the exception message makes no sense

Reply:
I think it does, as the method could be invoked with a `TUMCabeVerification` that has no `data` set. In this case, `IllegalStateException` seems like exactly the right exception to me. 

How would you handle it?

Reply:
C++ templates would be much better here. To properly handle it you would probably use https://stackoverflow.com/questions/7526817/use-of-instance-of-in-java

You can still check for a null value even before the cast and then throw the illegalstate.